api/handler/task.go

```diff
@@ -270,9 +271,13 @@ func UpdateTask(c *gin.Context, in *updateTaskIn) (*task.Task, error) {
 	t.SetInput(clearInput)
 	t.SetWatcherUsernames(in.WatcherUsernames)
 
-	if err := utils.ValidateTags(in.Tags); err != nil {
-		return nil, err
+	// validate read-only tags
+	v, readOnlyTagUpdated := in.Tags[constants.SubtaskTagParentTaskID]
+	oldValue, readOnlyTagInTask := t.Tags[constants.SubtaskTagParentTaskID]
+	if (readOnlyTagUpdated && (!readOnlyTagInTask || oldValue != v)) || (!readOnlyTagUpdated && readOnlyTagInTask) {
```

Comment:
Not sure about the second part of the condition:`(!readOnlyTagUpdated && readOnlyTagInTask)`. This implies that during an update, the read-only tag must be given and be the same value. Is it intentional ?

Reply:
It seems right to me, their absence would mean a deletion otherwise?

Reply:
Exactly :+1: